src/pages/AddJob.jsx

```diff
@@ -0,0 +1,182 @@
+import "react-datepicker/dist/react-datepicker.css";
+
+import { useRequest } from "ahooks";
+import { doc } from "firebase/firestore";
+import React from "react";
+import DatePicker from "react-datepicker";
+import { Controller, useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+import database from "@/clients/firebase";
+import { useAuth, useJob } from "@/models";
+import { englishLevel } from "@/utils/constants";
+
+export default function AddJob() {
+  const auth = useAuth();
+  const job = useJob();
+  const navigate = useNavigate();
+  const {
+    control,
+    register,
+    handleSubmit,
+    // formState: { errors }
+  } = useForm();
+  console.log(job);
+  const { run: addJob } = useRequest(
+    async (data) =>
+      job.createJob({
+        ...data,
+        company: auth.user.company,
+        owner: auth.userRef,
+        status: "open",
+        postDate: new Date(),
+        // fixme: temp solution: we need an admin to approve the job
+        admin: doc(database, "Users", "iKGlSJEUkWQjCHZMQhgrVixVAt42"),
+      }),
+    {
+      manual: true,
+      onSuccess: () => {
+        navigate("/");
+      },
+      onError: () => {
+        //todo: handle error
+        //   console.error(error);
+      },
+    }
+  );
+
+  return (
+    <div className="flex flex-col justify-center items-center">
+      <form onSubmit={handleSubmit(addJob)}>
+        <div className="form-control">
+          <label className="label" htmlFor="title">
+            Title
+          </label>
+          <input
+            type="text"
+            className="input input-bordered"
+            {...register("title", { required: true })}
+          />
+        </div>
+        <div className="form-control">
+          <label className="label" htmlFor="company">
+            Company Name
+          </label>
+          <input
+            type="text"
+            className="input input-bordered"
+            {...register("company")}
+            disabled
+          />
+        </div>
+        <div className="form-control">
+          <label className="label" htmlFor="startDate">
+            Start Date
+          </label>
+          <div className="input-group">
+            <div
+              className="relative max-w-sm"
+              style={{ zIndex: "9999", left: "17rem", top: "-0.5rem" }}
+            >
+              <div className="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none">
```

Comment:
it raises a question here. you will find that the design of the date picker is inconsistent with the rest of the website. should have way to fix this in the late sprint if we have time. but it is good for now.

Reply:
yeah, will fix it in later sprint